aiotfm/room.py

```diff
@@ -192,10 +205,33 @@ def from_packet(cls, packet: Packet):
 				player_count = packet.read16()
 				limit = packet.read8()
 				is_funcorp = packet.readBool()
-
-				rooms.append(RoomEntry(
-					name, language, country, player_count,
-					limit=limit, is_funcorp=is_funcorp
-				))
+				is_modified = packet.readBool()
+
+				# Read the modified properties
+				if is_modified:
+					shaman_skills = not packet.readBool()
+					consumables = not packet.readBool()
+					adventure = not packet.readBool()
+					collision = packet.readBool()
+					aie = packet.readBool()
+					map_duration = packet.read8()
+					mice_mass = packet.read32()
+					map_rotation = []
+					for i in range(0, packet.read8()):
+						map_rotation.append(packet.read8())
+
+					rooms.append(RoomEntry(
+						name, language, country, player_count,
+						limit=limit, is_funcorp=is_funcorp, is_modified=is_modified,
+						shaman_skills=shaman_skills, consumables=consumables,
+						adventure=adventure, collision=collision, aie=aie,
+						map_duration=map_duration, mice_mass=mice_mass,
+						map_rotation=map_rotation
+					))
+				else:
+					rooms.append(RoomEntry(
+						name, language, country, player_count,

```

Comment:
I feel like this is repeating many lines. A better approach to this would be creating the a `kwargs` dictionary before creating the room instance, with all the arguments that are present both with a modified room or unmodified. If the room is modified, you update that dictionary with the new values and then pass that to the `RoomEntry` class. Example:

```python
kwargs = {
  "limit": limit,
  "is_funcorp": is_funcorp,
}

if is_modified:
  # unpack values...
  kwargs.update({
    "is_modified": True,
    "shaman_skills": shaman_skills,
    ...
  })

rooms.append(RoomEntry(
  name, language, country, player_count,
  **kwargs
))
```

Another approach would be to create the instance before checking if the room is modified, create in it a method `apply_modifications` and calling it with the proper modifications, if any.

Reply:
I agree. I will commit now.